classes/class-wc-connect-nux.php

```diff
@@ -83,5 +91,272 @@ public function register_add_service_to_zone_pointer( $pointers ) {
 			);
 			return $pointers;
 		}
+
+		public function get_jetpack_install_status() {
+			// check if Jetpack is activated
+			if ( ! class_exists( 'Jetpack_Data' ) ) {
+				// not activated, check if installed
+				if ( 0 === validate_plugin( 'jetpack/jetpack.php' ) ) {
+					return self::JETPACK_INSTALLED;
+				}
+				return self::JETPACK_UNINSTALLED;
+			} else if ( defined( 'JETPACK_DEV_DEBUG' ) && true === JETPACK_DEV_DEBUG ) {
+				// installed, activated, and dev mode on
+				return self::JETPACK_DEV;
+			}
+
+			// installed, activated, dev mode off
+			// check if connected
+			$user_token = Jetpack_Data::get_access_token( JETPACK_MASTER_USER );
+			if ( isset( $user_token->external_user_id ) ) { // always an int
+				return self::JETPACK_CONNECTED;
+			}
+
+			return self::JETPACK_ACTIVATED;
+		}
+
+		private function should_display_nux_notice_on_screen( $screen ) {
+			if ( // Display if on any of these admin pages.
+				( // Products list.
+					'product' === $screen->post_type
+					&& 'edit' === $screen->base
+				)
+				|| ( // Edit product page if at least one product already published
+					$this->get_number_of_published_products() > 0
+					&& 'product' === $screen->post_type
+					&& 'post' === $screen->base
+				)
+				|| ( // Orders list.
+					'shop_order' === $screen->post_type
+					&& 'edit' === $screen->base
+					)
+				|| ( // Edit order page.
+					'shop_order' === $screen->post_type
+					&& 'post' === $screen->base
+					)
+				|| ( // WooCommerce settings.
+					'woocommerce_page_wc-settings' === $screen->base
+					)
+				|| ( // WooCommerce featured extension page
+					'woocommerce_page_wc-addons' === $screen->base
+					&& isset( $_GET['section'] ) && 'featured' === $_GET['section']
+					)
+				|| ( // WooCommerce shipping extension page
+					'woocommerce_page_wc-addons' === $screen->base
+					&& isset( $_GET['section'] ) && 'shipping_methods' === $_GET['section']
+					)
+				|| 'plugins' === $screen->base
+			) {
+				return true;
+			}
+			return false;
+		}
+
+		private function get_number_of_published_products() {
+			$count = wp_count_posts( 'product' );
+			return $count->publish;
+		}
+
+		private function is_new_store() {
+			$order_count = 0;
+			$posts_by_status = wp_count_posts( 'shop_order' );
+			foreach ( $posts_by_status as $post_status => $count ) {
+				if ( in_array( $post_status, array(
+					'wc-pending',
+					'wc-processing',
+					'wc-on-hold',
+					'wc-completed',
+					'wc-cancelled',
+					'wc-refunded',
+					'wc-failed',
+				), true ) ) {
+					$order_count += $count;
+				}
+			}
+			return 0 === $order_count;
+		}
+
+		private function get_jetpack_redirect_url() {
+			$full_path = add_query_arg( array() );
+			// Remove [...]/wp-admin so we can use admin_url().
+			$new_index = strpos( '/wp-admin', $full_path ) + strlen( '/wp-admin' );
+			$path = substr( $full_path, $new_index );
+			return admin_url( $path );
+		}
```

Comment:
Closing the loop, [the conversation continues here](https://github.com/Automattic/woocommerce-services/pull/1031#discussion_r122555461).